scipy/interpolate/_fitpack_repro.py

```diff
@@ -139,7 +139,8 @@ def _validate_inputs(x, y, w, k, s, xb, xe, parametric):
 
 
 def generate_knots(x, y, *, w=None, xb=None, xe=None, k=3, s=0, nest=None):
-    """Replicate FITPACK's constructing the knot vector.
+    """
+    Generate the knot vector of a spline satisfying the Least SQuares (LSQ) criterion.
```

Comment:
How about "Generate knot vectors until the LSQ criterion is satified."

Reply:
I didn't see that this *was* a Python generator. I'm curious - why doesn't it just return the last one?

Reply:
Yup, it was added recently for https://github.com/scipy/scipy/issues/2579

The North Star kind of a goal is to allow a power user to mix and match 1) knot placement strategies, and 2) optimization criteria: 3rd derivative jumps (splrep), integral of the 2nd derivative (make_smoothing_spline), P-splines (not implemented in scipy).